README.md

```diff
@@ -1,6 +1,192 @@
+
+You can implement the Asset interface to supply any byte-based content as an InputStream. 
+For example, you can present a DataSource as an Asset:
+```java
+final DataSource dataSource = null; // Assume you have this
+Asset asset = new Asset() {
+  @Override
+  public InputStream openStream() {
+    try {
+      return dataSource.getInputStream();
+    } catch (final IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+};
+```
+
+You can switch views of your archive using the as method from the Assignable interface.
+For example, to add classes to a JAR file, use the JavaArchive view:
+
+```java
+myArchive.as(JavaArchive.class).addClasses(String.class, Integer.class);
+```
+
+#### Working with File Content
+
+ShrinkWrap provides functionality to package file content.
+Following example shows how to add all of the `.class` files in the current package.
+
+```java
+JavaArchive archive = ShrinkWrap.create(JavaArchive.class, "myPackage.jar")
+  .addPackage(this.getClass().getPackage());
+```
+
+Then, we can export the `JavaArchive` as ZIP, using property of `Assignable` interface, to view it as `ZipExporter`.
+
+```java
+archive.as(ZipExporter.class).exportTo(new File("/home/user/Desktop/myPackage.jar"), true);
+```
+
+Shrinkwrap allows exporting to `File`, `OutputStream` and `InputStream` and provides `TAR`, `TAR.GZ`, `ZIP` and `Exploded` exporters.
+
+```java
+// retrieves one of the exporter classes
+final Class<? extends StreamExporter> exporter = this.getExporterClass();
+// views the archive as exporter class and exports contents to target file
+archive.as(exporter).exportTo(File target,  boolean overwrite);
+```
+
+ShrinkWrap can also obtain an `Archive` from a flat-file using one of the importers, which mirror their respective exporters.
+
+```java
+JavaArchive roundtrip = ShrinkWrap
+  .create(ZipImporter.class, "myPackageRoundtrip.jar")
+  .importFrom(new File("/home/user/Desktop/myPackage.jar"))
+  .as(JavaArchive.class);
+// or you can use more simple way for ZIP files
+JavaArchive archive = ShrinkWrap.createFromZipFile(JavaArchive.class, new File("/home/user/Desktop/myPackage.jar"));
+```
+
+### Domains
+
+In ShrinkWrap, a `Domain` encapsulates a configuration context for creating archives.
+Different domains can have isolated configurations, allowing you to manage multiple environments within a single application.
+
+
+#### Creating Domains
+
+You can create a new domain with a default configuration, a custom configuration, or a custom configuration builder.
+
+The `Configuration` class in ShrinkWrap encapsulates all the settings and properties for a given `Domain`.
+Each `Archive` created by the domain's `ArchiveFactory` will use this configuration internally.
+
+```java
+// Create a domain with default configuration
+Domain defaultDomain = ShrinkWrap.createDomain();
+
+// Create a domain with a custom configuration builder
+Domain domain = ShrinkWrap.createDomain(new ConfigurationBuilder().executorService(service)
```

Comment:
Perhaps an example about the executorService or extensionLoader could be provided as well. If it's short enough

Reply:
I don't think I can write any short example..., but there are more extensive examples in tests.